src/index.ts

```diff
@@ -29,6 +35,56 @@ export const pluginPug = (options: PluginPugOptions = {}): RsbuildPlugin => ({
 			config: options.pugOptions,
 		});
 
+		const state: {
+			readyToReload: boolean;
+			server?: SetupMiddlewaresServer;
+		} = {
+			readyToReload: false,
+			server: undefined,
+		};
+
+		/**
+		 * Trigger a page reload once on .pug entry rebuild
+		 */
+		const triggerPageReload = debounce(
+			() => {
+				state.server?.sockWrite('static-changed');
+			},
+			100,
+			{
+				leading: false,
+				trailing: true,
+			},
+		);
+
+		// Setup middleware to get access to the dev server instance.
+		api.modifyRsbuildConfig((config) => {
```

Comment:
Sorry I forget to mention that Rsbuild allows you to use the [onBeforeStartDevServer](https://rsbuild.dev/plugins/dev/hooks#onbeforestartdevserver) hook to register middleware, which is easier than using `modifyRsbuildConfig`.

See: https://rsbuild.dev/guide/basic/server#register-middlewares